cogs/countdown.py

```diff
@@ -1,31 +1,54 @@
 import logging
 import datetime
 import asyncio
+import random
 from discord.ext import tasks, commands
-from constants import ANNOUNCEMENTS_CHAN_ID
+from constants import ANNOUNCEMENTS_CHAN_ID, HYPE_MSGS
 
 logger = logging.getLogger(__name__)
 
 
 class Countdown(commands.Cog):
     def __init__(self, bot):
         self.bot = bot
-        year = datetime.datetime.now().year
-        self.bitcoin_pizza_day = datetime.date(year, 5, 22)  # May 22, 2021 etc
+        year = datetime.datetime.utcnow().year
+        self.bitcoin_pizza_day = datetime.date(year, 5, 22)
+        self.tau_day = datetime.date(year, 6, 28)
         self.target_hour = 15  # 3pm UTC / 11am EDT / 10am CDT / 8am PDT
-        self.message = "Hello @everyone!! Today is {date}, which means there are **{days} DAYS until May 22**, Bitcoin Pizza Day! Let's goooo!"
+        self.messages = {
+            "0522": "Hello @everyone! Today is {date}, which means there are **{days} DAYS until May 22nd**, Bitcoin Pizza Day! {hype}",
+            "0628": "Hello @everyone! Today is {date}, which means there are **{days} DAYS until June 28th**, Tau Day! {hype}",
+            "default": "Oops, looks like my circuits got mixed up. Today is {date}; days is {days}; hype message is {hype}. Haaalp debug me!",
+        }
         self.announcement.start()
 
     def cog_unload(self):
         self.announcement.cancel()
 
     @tasks.loop(hours=24)
     async def announcement(self):
+        # bitcoin pizza day, tau day, or next year
         today = datetime.date.today()
-        days = (self.bitcoin_pizza_day - today).days
+        target_day = datetime.date(1959, 10, 31)
+        if today <= self.bitcoin_pizza_day:
+            target_day = self.bitcoin_pizza_day
+        elif today <= self.tau_day:
+            target_day = self.tau_day
+        else:
+            target_day = datetime.date(today.year + 1, 5, 22)  # next year
```

Comment:
This wont work between January and May

Reply:
just the else or the entire if/elif/else logic? 

this is how i currently see it working: 
if it's `Jan 01 2022`, it will fall into the if statement b/c `01/01/2022 <= 05/22/2022` and so target day is bitcoin pizza day aka `05/22/2022`
then later on when we pass that day, say it's `05/23/2022`, it'll skip the `if` and fall into `elif` since `05/23/2022` < `06/28/2022` so target day is tau day aka `06/28/2022`. otherwise we have passed both bitcoin pizza day and tau day and so we look forward to next year's bitcoin pizza day

Reply:
I was gonna say something different. I think this function is actually not generic enough. There really should just be a list of events that is passed in and it determines the next date. 

```
event = get_next_event()
target_day = event.date
```